reV/SAM/generation.py

```diff
@@ -624,6 +625,337 @@ def default(self):
+            are shifted to relabel Feb 29th as March 1st, March 1st as March
+            2nd, etc.
+        """
+        self._drop_leap = drop_leap
+        super().__init__(resource=resource, meta=meta, parameters=parameters,
+                         output_request=output_request, drop_leap=False)
+
+    def set_nsrdb(self, resource):
+        """
+        Set NSRDB resource file. Overloads Solar.set_nsrdb(). Solar thermal
+        PySAM models require a data file, not raw data.
+
+        Parameters
+        ----------
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+        """
+        self.time_interval = self.get_time_interval(resource.index.values)
+        self._pysam_w_fname = self._create_pysam_wfile(self._meta, resource)
+        # pylint: disable=E1101
+        self[self._pysam_weather_tag] = self._pysam_w_fname
+
+    def _create_pysam_wfile(self, meta, resource):
+        """
+        Create PySAM weather input file. PySAM will not accept data on Feb
+        29th. For leap years, December 31st is dropped and time steps are
+        shifted to relabel Feb 29th as March 1st, March 1st as March 2nd, etc.
+
+        Parameters
+        ----------
+        meta : pd.DataFrame
+            1D table with resource meta data.
+        resource : pd.DataFrame
+            2D table with resource data. Available columns must have var_list.
+
+        Returns
+        -------
+            fname : string
+                Name of weather csv file
+        """
+        fname = '{}_weather.csv'.format(self._site)
+        logger.debug('Creating PySAM weather data file: {}'.format(fname))
+
+        # ------- Process metadata
+        m = pd.DataFrame(meta).T
+        timezone = m.timezone
+        m['Source'] = 'NSRDB'
+        m['Location ID'] = meta.name
+        m['City'] = '-'
+        m['State'] = m.state.apply(lambda x: '-' if x == 'None' else x)
+        m['Country'] = m.country.apply(lambda x: '-' if x == 'None' else x)
+        m['Latitude'] = m.latitude
+        m['Longitude'] = m.longitude
+        m['Time Zone'] = m.timezone
+        m['Elevation'] = m.elevation
+        m['Local Time Zone'] = m.timezone
+        m['Dew Point Units'] = 'c'
+        m['DHI Units'] = 'w/m2'
+        m['DNI Units'] = 'w/m2'
+        m['Temperature Units'] = 'c'
+        m['Pressure Units'] = 'mbar'
+        m['Wind Speed'] = 'm/s'
+        m = m.drop(['elevation', 'timezone', 'country', 'state', 'county',
+                    'urban', 'population', 'landcover', 'latitude',
+                    'longitude'], axis=1)
+        m.to_csv(fname, index=False, mode='w')
+
+        # --------- Process data
+        df = resource.copy()
+        df['dt'] = df.index
+
+        # Adjust from UTC to local time
+        rolled = np.roll(df.to_numpy(), timezone * self.time_interval,
+                         axis=0)
+        df = pd.DataFrame(rolled, columns=df.columns, index=df.index)
```

Comment:
Okay so rolling everything together here is good, but what I would do is make DF as a blank dataframe with index as the resource.index and always drop the leap day if present. Then you use ensure_res_len() when you're setting 1D arrays to the columns in DF.  I think that would be a cleaner way to handle leap days. The stuff you've got going on down at 752 is super confusing